### PR TITLE
Implement updated header designs across Sourcegraph pages

### PR DESCRIPTION
**Pull request by umpox**
This PR:
- Updates styling to match the new Header on the following pages
  - Extensions / Extension details
    - This also involves moving the Extension icon to the sidebar, and supporting the default icon for Sourcegraph-authored extensions
  - Campaigns / Campaign details
  - Profile / Org pages
  - Insights page
  - Query builder

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
campaigns code changes LGTM

File: client/web/src/user/area/UserAreaHeader.tsx [hunk 1]
```diff
@@ -25,17 +26,19 @@ export const UserAreaHeader: React.FunctionComponent<Props> = ({ url, navItems,
...
-                        ) : (
-                            props.user.username
-                        )}
-                    </h2>
+                    <PageHeader
+                        path={[
+                            {
+                                text: (
+                                    <span className="align-middle">
+                                        {props.user.displayName || props.user.username}
```
Comment (umpox) [inline client/web/src/user/area/UserAreaHeader.tsx:34]: @AlicjaSuska We currently display the username _and_ display name if they are both available. From the new designs I am assuming that we only want to display one value: the display name of the user where possible and username when not. The new way also matches what the org header does. Let me know if this assumption is wrong!

File: client/web/src/enterprise/campaigns/list/CampaignListPage.tsx [hunk 2]
```diff
@@ -120,7 +120,6 @@ export const CampaignListPage: React.FunctionComponent<CampaignListPageProps> =
                 path={[{ icon: CampaignsIcon, text: 'Campaigns' }]}
                 className="test-campaign-list-page mb-3"
                 actions={<NewCampaignButton location={location} />}
-                byline="Run custom code over hundreds of repositories and manage the resulting changesets"
```
Comment (umpox) [inline client/web/src/enterprise/campaigns/list/CampaignListPage.tsx:123]: @AlicjaSuska The designs didn't include this copy so I have removed it from the Campaign header, let me know if this was not intended 😄

File: client/web/src/extensions/extension/ExtensionAreaHeader.tsx [hunk 3]
```diff
@@ -88,70 +68,61 @@ export const ExtensionAreaHeader: React.FunctionComponent<ExtensionAreaHeaderPro
...
-                            <div className="mb-3">
-                                <div className="d-flex align-items-start">
-                                    {iconURL && (
-                                        <img
-                                            className="extension-area-header__icon mr-2"
-                                            src={iconURL.href}
-                                            aria-hidden="true"
-                                        />
+                        <PageHeader
+                            annotation={
```
[the remark this answers is not in the document]
Comment (AlicjaSuska) [reply client/web/src/extensions/extension/ExtensionAreaHeader.tsx:75]: LGTM thank you @umpox the idea is to move all the labels (also the 'WIP' one) above the header.

File: client/web/src/user/area/UserAreaHeader.tsx (hunk 1 above)
Comment (AlicjaSuska) [reply client/web/src/user/area/UserAreaHeader.tsx:34]: I think we should keep both, thank you for spotting this detail. How about '[Display name (username)](https://www.figma.com/file/A4gGoseJDz8iPeHP515MfQ/%F0%9F%A5%96Breaders-breadcrumbs-headers-15431-Approved?node-id=1210%3A11457)':
Screenshot 2021-01-29 at 16 14 11

File: client/web/src/enterprise/campaigns/list/CampaignListPage.tsx (hunk 2 above)
Comment (AlicjaSuska) [reply client/web/src/enterprise/campaigns/list/CampaignListPage.tsx:123]: All the bylines copy should stay as they are. So the one mentioned above and also, for the campaigns starting page 'Make and track large-scale changes across all code' (as displayed currently):
Screenshot 2021-01-29 at 16 16 45

File: client/web/src/user/area/UserAreaHeader.tsx (hunk 1 above)
Comment (umpox) [reply client/web/src/user/area/UserAreaHeader.tsx:34]: Yeah that works! Thank you!

File: client/web/src/enterprise/campaigns/list/CampaignListPage.tsx (hunk 2 above)
Comment (umpox) [reply client/web/src/enterprise/campaigns/list/CampaignListPage.tsx]: Fixed!

File: client/web/src/user/area/UserAreaHeader.tsx (hunk 1 above)
Comment (umpox) [reply client/web/src/user/area/UserAreaHeader.tsx]: Have updated 😄

**Comment by AlicjaSuska** [comment]
@umpox Can we add 'Improve your workflow...' text on the extensions registry page as header byline as well?
Screenshot 2021-02-01 at 12 21 20

Also, I've noticed that when using the default avatar, the image in the header is too small. It should have the same size as for a custom avatar.

Screenshot 2021-02-01 at 12 21 46

**Comment by AlicjaSuska** [comment]
Looks good, thank you for making those changes @umpox ! All ready to go from my part 🎉

**Review by AlicjaSuska — APPROVED** [review]

**Review by tjkandala — APPROVED** [review]
Extensions changes look good!

**Pull request merged**